docs/hello_nextflow/06_hello_config.md

```diff
@@ -0,0 +1,266 @@
+# Part 5: Hello Config
+
+In this section, we will explore how to configure Nextflow pipelines using configuration files, profiles, process directives, and executors. Configuration management is an essential aspect of Nextflow pipeline development, allowing you to customize the behavior of your pipeline, adapt it to different environments, and optimize resource usage. By understanding and utilizing these configuration options effectively, you can enhance the flexibility, scalability, and performance of your pipelines.
+
+## 1. Check and modify configuration
+
+### 1.1. Run nf-hello-gatk with default settings
+
+```bash
+nextflow run seqeralabs/nf-hello-gatk
+```
+
+When you run the pipeline with the default settings using the command above, the following happens:
+
+1. Nextflow downloads the pipeline from the GitHub repository `seqeralabs/nf-hello-gatk`.
+2. It then executes the pipeline using the default configuration.
+3. The pipeline will likely use Docker containers to run the required tools (Samtools and GATK).
+4. It processes the input BAM files, creating index files and performing variant calling.
+5. The results are generated in the `results` directory by default.
+6. Nextflow also creates a `work` directory containing intermediate files and logs.
+7. Upon completion, Nextflow displays a summary of the run, including any errors or warnings.
+
+Now let's move on to seeing how this was configured and set up.
+
+### 1.2. Check configuration
+
+Open the `nextflow.config` file and inspect the contents:
+
+```bash
+code nextflow.config
+```
+
+The contents should look like this:
+
+```groovy title="nextflow.config" linenums="1"
+docker.enabled = true
+```
+
+This means that the pipeline will use Docker containers to run the required tools.
+
+### 1.3. Modify configuration
+
+Let's modify the configuration to use Conda instead of Docker.
+
+_Before:_
+
+```groovy title="nextflow.config" linenums="1"
+docker.enabled = true
+```
+
+_After:_
+
+```groovy title="nextflow.config" linenums="1"
+docker.enabled = false
+conda.enabled = true
+```
+
+Now let's run the pipeline again with the modified configuration:
+
+```bash title="nextflow.config"
+nextflow run seqeralabs/nf-hello-gatk
+```
+
+This time, the pipeline will use Conda environments to run the required tools.
+
+## 2. Profiles
+
+Profiles are a way to customize the behavior of Nextflow pipelines by selection, rather than setting them permanently.
+
+### 2.1. Create a profile
+
+_Before:_
+
+```groovy title="nextflow.config" linenums="1"
+docker.enabled = false
+conda.enabled = true
+```
+
+_After:_
+
+```groovy title="nextflow.config" linenums="1"
+profiles {
+    'docker' {
+        docker.enabled = true
+        conda.enabled = false
```

Comment:
I think the enabled = false are overkill.
We have that in nf-core, but we might be doing too much